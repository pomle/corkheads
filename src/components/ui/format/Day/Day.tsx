import React from "react";
import { Moment } from "moment";

const FORMAT = "YYYY-MM-DD";

interface DayProps {
  date: Moment;
}

const Day: React.FC<DayProps> = ({ date }) => {
  return <>{date.format(FORMAT)}</>;
};

export default Day;
