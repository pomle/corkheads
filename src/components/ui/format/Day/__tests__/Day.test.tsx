import React from "react";
/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import renderer from "react-test-renderer";
import moment from "moment";
import Day from "../Day";

describe("Day component", () => {
  const VALID_MOMENT = moment("2019-09-12T12:42:13.423Z");

  it("shows the day of date when valid", () => {
    const component = renderer.create(<Day date={VALID_MOMENT} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
