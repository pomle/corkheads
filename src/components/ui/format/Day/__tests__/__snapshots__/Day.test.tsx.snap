// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component shows the day of date when valid 1`] = `"2019-09-12"`;
